feat(curriculum): forum leaderboard project (#50241)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/645f764c37de33015ded1273.md

---
id: 645f764c37de33015ded1273
title: Step 105
challengeType: 0
dashedName: step-105
---

# --description--

> The `user` object's property `avatar_template` is a string that contains a placeholder `{size}`. You will create a constant named `avatar` and assign it the value of the `user` object's property `avatar_template` with the placeholder `{size}` replaced with the desired size of the avatar, in this case, `30`. This will allow you to customize the size of the user's avatar in your application.

Reviewer: It might be a bit hard to understand what exactly is asked. Maybe including a full example of the `avatar_template` string, would make it easier.

# --hints--

> You should create a constant named `avatar` and assign it the value of the `user` object's property `avatar_template` with the placeholder `{size}` replaced with `30`.

Reviewer: ```suggestion
Replace the `{size}` placeholder, in the `avatar_template` property of `user` object, with `30`. Then assign it to the `avatar` constant.
```
Shifted wording a bit, to put the main task earlier.

Author: added

```js

> assert(code.match(/const\s+avatar\s*=\s*user\.avatar_template\.replace\(\s*\/{size}\/\s*,\s*30\s*\)/));

Reviewer: ```suggestion
assert(code.match(/const\s+avatar\s*=\s*user\.avatar_template\.replace\(\s*([/'"]){size}\1\s*,\s*30\s*\)/));
```
Test is expecting to use regex, but string will work in here in the same way.

Author: added
